### ISSUE TITLE
Function request: np.isin

### ISSUE THREAD

**Issue by aron-bordin** [pytorch/pytorch#3025]
I believe that should be useful to have a function similar to https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.in1d.html, that compares a tensor element-wise with a list of possible values. (I'm using it to filter labels/classes in some classifiers).

Expected behavior:

```
>>> a = torch.LongTensor([[1,2,3],[1,1,2],[3,5,1]])
>>> a
 1  2  3
 1  1  2
 3  5  1
[torch.LongTensor of size 3x3]
>>> a.in(torch.LongTensor([1, 2, 5]))
 1 1 0
 1 1 1
 0 1 1
```

Now it's possible to implement it by iterating over the filter, and storing the results in a tensor with the OR operator. But a faster implementation is possible with TH/THC

My current implementation:

```
@utils.tensorfy(0, 1, tensor_klass=torch.LongTensor)
def filter_labels(y, labels):
    """Utility used to create a mask to filter values in a tensor.

    Args:
        y (list, torch.Tensor): tensor where each element is a numeric integer
            representing a label.
        labels (list, torch.Tensor): filter used to generate the mask. For each
            value in ``y`` its mask will be "1" if its value is in ``labels``,
            "0" otherwise".

    Shape:
        y: can have any shape. Usually will be :math:`(N, S)` or :math:`(S)`,
            containing `batch X samples` or just a list of `samples`.
        labels: a flatten list, or a 1D LongTensor.

    Returns:
        mask (torch.ByteTensor): a binary mask, with "1" with the respective value from ``y`` is
        in the ``labels`` filter.

    Example::

        >>> a = torch.LongTensor([[1,2,3],[1,1,2],[3,5,1]])
        >>> a
         1  2  3
         1  1  2
         3  5  1
        [torch.LongTensor of size 3x3]
        >>> classification.filter_labels(a, [1, 2, 5])
         1  1  0
         1  1  1
         0  1  1
        [torch.ByteTensor of size 3x3]
        >>> classification.filter_labels(a, torch.LongTensor([1]))
         1  0  0
         1  1  0
         0  0  1
        [torch.ByteTensor of size 3x3]
    """
    mapping = torch.zeros(y.size()).byte()

    for label in labels:
        mapping = mapping | y.eq(label)

    return mapping

```

cc @mruberry @rgommers @heitorschueroff

**Comment by nishnik** [comment]
If the issue is still open, Can I take this up?

**Comment by fmassa** [comment]
Sure! Also, instead of implementing `in1d`, I think we should implement the more general `isin`, as recommended by [the numpy documentation](https://docs.scipy.org/doc/numpy/reference/generated/numpy.in1d.html)
Here is a slightly more runtime efficient (but memory-hungry and suboptimal) implementation (requires PyTorch compiled from master):
```python
def isin(ar1, ar2):
    return (ar1[..., None] == ar2).any(-1)
```

**Comment by Wangbicong** [comment]
I think this interface is very useful. 
So I hope it can be added as soon as possible.

**Comment by ZaydH** [comment]
@CamiWilliams When I run your proposed `isin` implementation with `torch` 1.2.0 on MacOS 10.14.6, I get the warning:

`../aten/src/ATen/native/IndexingUtils.h:20: UserWarning: indexing with dtype torch.uint8 is now deprecated, please use a dtype torch.bool instead.`

If you change:

`return result.type(torch.ByteTensor)`

to:

`return result.type(torch.BoolTensor)`

the warning goes away.  Could you provide more detail on why you went with `ByteTensor` here?  I want to make sure I did not miss something.

**Comment by divyanshj16** [comment]
When will `torch.isin` functionality be added in pytorch or is there already something like this which is not documented?
https://github.com/pytorch/pytorch/pull/26144

**Comment by glaringlee** [comment]
assign to myself for now. This might be a good task for onboading team member.

**Comment by mberr** [comment]
Another work-around for `in1d` which is reasonably fast; memory consumption depends on maximum element of `a1` and `ar2`.

```python
def in1d(ar1, ar2):
    mask = ar2.new_zeros((max(ar1.max(), ar2.max()) + 1,), dtype=torch.bool)
    mask[ar2.unique()] = True
    return mask[ar1]
```

**Comment by mruberry** [comment]
Quick fyi: following NumPy's recommendation we should implement isin to start.

**Comment by mberr** [comment]
Although it looks like numpy is currently just wrapping `in1d`:

https://github.com/numpy/numpy/blob/da887a666ad975ece7fb7465005aa99c0ddef8d2/numpy/lib/arraysetops.py#L709-L711

**Comment by ShirAmir** [comment]
any news on this feature? super useful

**Comment by mruberry** [comment]
> any news on this feature? super useful

Yes the function has been added (which is why this issue is closed). See https://pytorch.org/docs/master/generated/torch.isin.html?highlight=isin#torch.isin
